components/account/home/AccountInformation.vue

```diff
@@ -93,10 +98,116 @@ async function update() {
   set(loading, false);
 }
 
+const waitTime = ref<number>(0);
+const loadingCheck = ref<boolean>(false);
+
+async function handleCheckVATClick() {
+  set(loadingCheck, true);
+  const result = await store.checkVAT();
+  await store.refreshVATCheckStatus();
+
+  if (typeof result === 'number') {
+    set(waitTime, result);
+  }
+  set(loadingCheck, false);
+}
+
 const { t } = useI18n();
+
+const timer = ref<NodeJS.Timeout | null>(null);
+
+watch(waitTime, (newValue) => {
+  // Clear any existing timer
+  const timerVal = get(timer);
+  if (timerVal) {
+    clearInterval(timerVal);
+  }
+
+  // If waitTime is greater than 0, start a countdown
+  if (newValue > 0) {
+    const newTimer = setInterval(() => {
+      const wait = get(waitTime);
+      if (wait > 0) {
+        set(waitTime, wait - 1);
+      }
+      else {
+        const timerVal = get(timer);
+        if (timerVal) {
+          clearInterval(timerVal);
+        }
+      }
+    }, 1000);
+    set(timer, newTimer);
+  }
+});
```

Comment:
I wonder if there is a better way to do the countdown

Reply:
It's simplified now

Reply:
Thank you :)